src/main/preload.ts

```diff
@@ -1,25 +1,35 @@
 import { contextBridge, ipcRenderer, IpcRendererEvent } from 'electron';
-import { IpcChannel, IpcChannelMessage } from '../common/ipc';
+import {
+  IpcChannel,
+  IpcChannelMessage,
+  IpcChannelResponse,
+} from '../common/ipcMain';
 
 contextBridge.exposeInMainWorld('electron', {
   ipcRenderer: {
     sendMessage<T extends IpcChannel>(
       channel: T,
       message: IpcChannelMessage<T>
     ) {
-      ipcRenderer.send(channel, [message]);
+      ipcRenderer.send(channel, message);
     },
     on<T extends IpcChannel>(
       channel: T,
       callback: (message: IpcChannelMessage<T>) => void
     ) {
-      const subscription = (event: IpcRendererEvent, args: unknown[]) => {
-        const message = args[0] as IpcChannelMessage<T>;
```

Comment:
for some reason when reading the ipc API docs I was always thinking args must be a list, but it can be anything.

Let's stop passing single-element lists and just pass the message instead :upside_down_face: